monitors.go

```diff
@@ -152,3 +152,50 @@ func (m *Monitor) Instances(key *aurora.JobKey, instances int32, interval int, t
 	fmt.Println("Timed out")
 	return false, nil
 }
+
+// Monitor host status until all hosts match the status provided. Returns a map where the value is true if the host
+// is in one of the desired mode(s) or false if it is not.
+func (m *Monitor) HostMaintenance(hosts []string, modes []aurora.MaintenanceMode, sleepTime, steps int) (map[string]bool, error) {
+
+	//  Transform modes to monitor for into a set for easy lookup
+	desiredMode := make(map[aurora.MaintenanceMode]struct{})
+	for _, mode := range modes {
+		desiredMode[mode] = struct{}{}
+	}
+
+	// Turn slice into a host set to eliminate duplicates.
+	observedHosts := make(map[string]bool)
+	for _, host := range hosts {
+		observedHosts[host] = false
+	}
+
+	for step := 0; step < steps; step++ {
+		// Client may have multiple retries handle retries
+		_, result, err := m.Client.MaintenanceStatus(hosts...)
+		if err != nil {
+			// Error is either a payload error or a severe connection error
+			return observedHosts, errors.Wrap(err, "client error")
+		}
+
+		for status := range result.GetStatuses() {
+			if _, ok := desiredMode[status.GetMode()]; ok {
+				observedHosts[status.GetHost()] = true
+
+				transitionedHosts := 0
+				for _, val := range observedHosts {
+					if val {
+						transitionedHosts++
+					}
+				}
+
+				if len(observedHosts) == transitionedHosts {
+					return observedHosts, nil
+				}
+			}
+		}
+
+		time.Sleep(time.Duration(sleepTime) * time.Second)
```

Comment:
The execution should be paused only if step < steps -1, otherwise at the last loop the pause will be useless

Reply:
Good catch.